### PR TITLE
Remove unused aws-centos-40 config map

### PR DESCRIPTION
**Pull request by psalajova**
This config map is not used in any test. A part of the initiative to clean up possibly deprecated derivative cluster profiles (profiles which share both secret and leases with the original aws/gcp/vsphere cluster profile).

/cc @jmguzik

### CODE REVIEW

File: core-services/prow/02_config/_plugins.yaml [hunk 1]
```diff
@@ -411,10 +411,6 @@ config_updater:
       cluster_groups:
       - build_farm_ci
       name: prow-job-cluster-launch-installer-openstack-upi-e2e
-    cluster/test-deploy/aws-4.0/*:
```
Comment (droslean) [inline core-services/prow/02_config/_plugins.yaml:414]: If this is not being used, we should assume that it is safe to delete those files as well.

Comment (psalajova) [reply core-services/prow/02_config/_plugins.yaml:414]: good point

**Review by droslean — APPROVED** [review]

**Pull request merged**